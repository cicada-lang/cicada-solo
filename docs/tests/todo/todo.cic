datatype Either(L: Type, R: Type) {
  inl(left: L): Either(L, R)
  inr(right: R): Either(L, R)
}

let a: String = TODO_NOTE("A trivial String")
let b: Trivial = TODO_NOTE("It's not Trivial")
let c: Absurd = TODO_NOTE("Absurd!")

function law_of_excluded_middle(A: Type): Either(A, (A) -> Absurd) {
  return TODO_NOTE("tertium non datur")
}

a
b
c
law_of_excluded_middle
