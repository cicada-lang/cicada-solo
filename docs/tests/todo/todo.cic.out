@TODO "A trivial String": String
sole: Trivial
the(Absurd, @TODO "Absurd!"): Absurd
(A) => @TODO "tertium non datur": (A: Type) -> Either(A, (_: A) -> Absurd)